ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClient.scala

```diff
@@ -36,4 +38,15 @@ final class EmberClient[F[_]] private[client] (
   def state: F[(Int, Map[RequestKey, Int])] = pool.state
 
   def run(req: Request[F]): Resource[F, Response[F]] = client.run(req)
+
+  override def defaultOnError(req: Request[F])(resp: Response[F])(implicit
+      G: Applicative[F]
+  ): F[Throwable] = F match {
+    case concurrentF: Concurrent[F @unchecked] =>
+      implicit val C = concurrentF
+      resp.body.compile.drain.as(UnexpectedStatus(resp.status, req.method, req.uri))
+
+    case _ =>
+      F.pure(UnexpectedStatus(resp.status, req.method, req.uri))
```

Comment:
Is this case worth logging a warning that the behavior is buggy?

Reply:
Basically, I believe this case shouldn't appear at runtime, because 99% of effect usage is CE.IO/monix.Task/etc effect that is supposed to have a `Concurrent` instance. But `defaultOnError` is used for serving requests overall, and end-users could even don't use logging, but this method actually will be used. So this case preserves the current behavior of request processing mostly, not just logging behavior.